variables.virtual.network.tf

```diff
@@ -0,0 +1,137 @@
+  default     = null
+  description = "The ID of a pre-existing subnet to use. Required if `create_virtual_network` is `false`."
+}
+
+variable "container_instance_subnet_name" {
+  type        = string
+  default     = null
+  description = "The name of the subnet. Must be specified if `create_virtual_network == false`."
+}
+
+variable "container_instance_subnet_cidr_size" {
+  type        = number
+  default     = 28
+  description = "The CIDR size for the container instance subnet."
+}
+
+variable "container_instance_subnet_address_prefix" {
+  type        = string
+  default     = null
+  description = "The address prefix for the Container App Environment. Either subnet_id or subnet_name and subnet_address_prefix must be specified."
+}
+
+variable "container_instance_subnet_id" {
+  type        = string
+  default     = null
+  description = "The ID of a pre-existing subnet to use. Required if `create_virtual_network` is `false`."
+}
+
+variable "container_registry_private_endpoint_subnet_name" {
+  type        = string
+  default     = null
+  description = "The name of the subnet. Must be specified if `create_virtual_network == false`."
+}
+
+variable "container_registry_subnet_cidr_size" {
+  type        = number
+  default     = 29
+  description = "The CIDR size for the container registry subnet."
+}
+
+variable "container_registry_private_endpoint_subnet_address_prefix" {
+  type        = string
+  default     = null
+  description = "The address prefix for the Container App Environment. Either subnet_id or subnet_name and subnet_address_prefix must be specified."
+}
+
+variable "container_registry_private_endpoint_subnet_id" {
+  type        = string
+  default     = null
+  description = "The ID of a pre-existing subnet to use. Required if `create_virtual_network` is `false`."
+}
+
+variable "create_container_registry_private_dns_zone" {
+  type        = bool
+  default     = true
+  description = "Whether or not to create a private DNS zone for the container registry."
+}
+
+variable "container_registry_dns_zone_id" {
+  type        = string
+  default     = null
+  description = "The ID of the private DNS zone to create for the container registry. Only required if `create_private_dns_zone` is `false`."
+}
+
+variable "create_public_ip" {
+  type        = bool
+  default     = true
+  description = "Whether or not to create a public IP."
+}
+
+variable "public_ip_name" {
+  type        = string
+  default     = null
+  description = "The name of the public IP."
+}
+
+variable "public_ip_id" {
+  type        = string
+  default     = null
+  description = "The ID of the public IP. Only required if `create_public_ip` is `false`."
+}
+
+variable "create_nat_gateway" {
+  type        = bool
+  default     = true
+  description = "Whether or not to create a NAT Gateway."
+}
+
+variable "nat_gateway_name" {
+  type        = string
+  default     = null
+  description = "The name of the NAT Gateway."
+}
+
+variable "nat_gateway_id" {
```

Comment:
Consider adding validation rules, now with TF 1.9 we can reference other vars

Reply:
Will come back to this for a future iteration. Will create an issue to track it.